File: medium/3653-xor-after-range-multiplication-queries-i/solution.py
from typing import List


class Solution:
    def xorAfterMultiplication(self, nums: List[int], queries: List[List[int]]) -> int:
        """
        3653. XOR After Range Multiplication Queries I
        Time: O(q × n)
        Space: O(1)
        """
        MOD = 10**9 + 7

        for l, r, k, v in queries:
            idx = l
            while idx <= r:
                nums[idx] = (nums[idx] * v) % MOD
                idx += k

        result = 0
        for num in nums:
            result ^= num

        return result

File: medium/3653-xor-after-range-multiplication-queries-i/solution.java
/**
 * 3653. XOR After Range Multiplication Queries I
 * Time: O(q × n)
 * Space: O(1)
 */
class Solution {
    public int xorAfterMultiplication(int[] nums, int[][] queries) {
        long MOD = 1_000_000_007L;

        for (int[] q : queries) {
            int l = q[0], r = q[1], k = q[2], v = q[3];

            int idx = l;
            while (idx <= r) {
                nums[idx] = (int) ((nums[idx] * (long) v) % MOD);
                idx += k;
            }
        }

        int result = 0;
        for (int num : nums) {
            result ^= num;
        }

        return result;
    }
}
